### PR TITLE
sync_diff: support json compare

### LINKED ISSUE #606: sync_diff_inspector 的 json compare 不正确
https://github.com/pingcap/tidb-tools/blob/master/sync_diff_inspector/utils/utils.go#L535

mysql 按照 key length 排序然后再按词典序，tidb 按照词典序排序，看起来这个地方把 json 解析再序列化下就可以比对了。

func CompareData(map1, map2 map[string]*dbutil.ColumnData, orderKeyCols, columns []*model.ColumnInfo) (equal bool, cmp int32, err error) {

image

### PR DESCRIPTION
**Pull request by xiongjiwei**
### What problem does this PR solve?

Issue Number: close #606 

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to be included in the release note

### CODE REVIEW

File: sync_diff_inspector/utils/utils.go [hunk 1]
```diff
@@ -533,6 +535,22 @@ func CompareData(map1, map2 map[string]*dbutil.ColumnData, orderKeyCols, columns
 			if math.Abs(num1-num2) <= 1e-6 {
 				continue
 			}
+		} else if column.FieldType.GetType() == mysql.TypeJSON {
+			if (str1 == str2) && (data1.IsNull == data2.IsNull) {
```
Comment (xiongjiwei) [inline sync_diff_inspector/utils/utils.go]:
```suggestion
			if (str1 == str2) || (data1.IsNull == data2.IsNull) {
```

**Comment by Leavrth** [comment]
could you add some integration tests?

the script of generating table is here,
https://github.com/pingcap/tidb-tools/blob/master/tests/sync_diff_inspector/run.sh#L20

**Review by 3pointer — APPROVED** [review]

**Review by Leavrth — APPROVED** [review]

**Pull request merged**